Makefile

```
@@ -280,6 +273,10 @@ init:
 bump:
 	git submodule foreach "git fetch origin&&git checkout master&&git reset --hard origin/master"
 
+deps:
+	mill -i _.resolvedIvyDeps
+	mill -i xiangshan.resolveFirtoolDeps
```

```suggestion
	mill -i xiangshan.resolveFirtoolDeps
	mill -i __.prepareOffline
```
Can we add the `prepareOffline` option somewhere so that some users can use the docker environment offline?

---

adding a `pull-image` target may meet this need?

```
+# pull docker image
+pull-image:
+       @docker pull $(XSDEV_IMAGE)
+
```

user can use `make pull-image && make verilog` to run chisel inside docker

**BUT**, we need to make sure c2d02fc38c6b70ce6020e2441e76c9c7ff78d159 works as expect first :)

---

If the purpose is to generate Verilog in an offline environment, the current workflows seems to be adequate. However, if compiling Chisel in an offline setting is necessary, additional setup steps might be required.

---

> prepareOffline

got it! https://github.com/com-lihaoyi/mill/issues/1576

maybe we can replace the whole things with this magic `prepareOffline`

---

sadly, we still need `mill -i xiangshan.resolveFirtoolDeps` to download llvm/firtool :(